crates/fuel-core/src/graphql_api/worker_service.rs

```diff
@@ -381,6 +386,64 @@ where
             )
             .into());
         }
+
+        for receipt in result.receipts() {
+            match receipt {
+                Receipt::Mint {
+                    sub_id,
+                    contract_id,
+                    val,
+                    ..
+                } => {
+                    let asset_id = contract_id.asset_id(sub_id);
+                    let current_count = match db.storage::<AssetsInfo>().get(&asset_id) {
+                        Ok(count) => count,
+                        Err(_) => {
+                            // If asset doesn't exist yet, create it with 0 count
+                            db.storage::<AssetsInfo>()
+                                .insert(&asset_id, &(*contract_id, **sub_id, 0))?;
+                            Some(Cow::Owned((*contract_id, **sub_id, 0)))
+                        }
+                    }
```

Comment:
I think we shouldn't treat all storage errors as "value not found". Some of them should just be bubbled up.

Maybe something like this will work:

```
                        let current_count = match db.storage::<AssetsInfo>().get(&asset_id)? {
                            Some(count) => Some(count),
                            None => {
                                // If asset doesn't exist yet, create it with 0 count
                                db.storage::<AssetsInfo>()
                                    .insert(&asset_id, &(*contract_id, **sub_id, 0))?;
                                Some(Cow::Owned((*contract_id, **sub_id, 0)))
                            }
                        }
```

If you agree, please note that now two branches always return `Some`, so maybe we can get rid of the follow-up `map()` as well.